### DISCUSSION TITLE
Breakdown by Region and City not working in self hosted option

### DISCUSSION THREAD

**Discussion by AtharvGraphy** [plausible/analytics#2287] [Self-Hosted Support] [answered]
I have self hosted plausible using docker. I have followed all the instructions mentioned in the docs. But when I try to get breakdown by region or city it returns empty result. 

Earlier when I used latest image from docker it was giving me error in the response after which I used stable version and then it started returning empty result instead of error. I have also tried using master tagged version but it also return empty response.

I checked on the UI, but it also doesn't show any data for region. But this works without any issue on the managed service.

What could be the issue here?

**Comment by AtharvGraphy** [comment]
When I send this request for breakdown by region on latest tagged docker version.
```
curl --location --request GET '{{BASE_URL}}/api/v1/stats/breakdown?site_id={{SITE_ID}}&property=visit:region&metrics=visitors' \
--header 'Authorization: Bearer {{AUTH_TOKEN}}'
```

I get this error on console
```
05:44:59.581 request_id=Fxp43DzrzvOWGaAB4OBB [warn] Failed to send Sentry event. Cannot send Sentry event because of invalid DSN
05:44:59.581 [error] #PID<0.8893.158> running PlausibleWeb.Endpoint (connection #PID<0.8851.158>, stream id 4) terminated
Server: 139.59.54.25:8000 (http)
Request: GET /api/v1/stats/breakdown?site_id=atharvgraphy.github.io&property=visit:region&metrics=visitors
** (exit) an exception was raised:
    ** (FunctionClauseError) no function clause matching in Plausible.Stats.Breakdown.do_group_by/2
        (plausible 0.0.1) lib/plausible/stats/breakdown.ex:250: Plausible.Stats.Breakdown.do_group_by(#Ecto.Query<from s0 in "sessions", where: s0.domain == ^"atharvgraphy.github.io", where: s0.timestamp >= ^~U[2022-09-02 18:30:00Z] and s0.start < ^~U[2022-10-03 18:30:00Z], order_by: [desc: fragment("uniq(?)", s0.user_id), asc: fragment("min(?)", s0.start)], limit: ^100, offset: ^0, select: %{}>, "visit:region")
        (plausible 0.0.1) lib/plausible/stats/breakdown.ex:188: Plausible.Stats.Breakdown.breakdown_sessions/5
        (plausible 0.0.1) lib/plausible_web/controllers/api/external_stats_controller.ex:68: PlausibleWeb.Api.ExternalStatsController.breakdown/2
        (plausible 0.0.1) lib/plausible_web/controllers/api/external_stats_controller.ex:1: PlausibleWeb.Api.ExternalStatsController.action/2
        (plausible 0.0.1) lib/plausible_web/controllers/api/external_stats_controller.ex:1: PlausibleWeb.Api.ExternalStatsController.phoenix_controller_pipeline/2
        (plausible 0.0.1) lib/plug/error_handler.ex:65: PlausibleWeb.Api.ExternalStatsController.call/2
        (phoenix 1.5.13) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2
        (plausible 0.0.1) lib/plausible_web/endpoint.ex:1: PlausibleWeb.Endpoint.plug_builder_call/2
```

and this in response
```
{
    "error": "** (FunctionClauseError) no function clause matching in Plausible.Stats.Breakdown.do_group_by/2"
}
```

**Comment by metmarkosaric** [comment, marked as answer]
Only the latest version is meant for self-hosting. You can find the latest version here: https://github.com/plausible/analytics/releases. These errors are expected if you're trying to use a version that we use for our cloud as those versions are not built for nor meant for self-hosting.

Regions and cities are not available on self-hosted yet. Self-hosted is a long-term release. We stack all the new features and fixes that have been battle-tested on the hosted version and release a new self-hosted version. We don't have any more specific timelines than this as we have limited resources. You can check out the features that self-hosted is [missing at the moment here](https://github.com/plausible/analytics/blob/master/CHANGELOG.md).

**Reply by AtharvGraphy** [reply]
thanks for reply

**Reply by ruslandoga** [reply]
@AtharvGraphy 👋

[`v1.5.0-rc.0`](https://github.com/plausible/analytics/discussions/2324) is out with [region / city level geolocation support.](https://github.com/plausible/analytics/discussions/2324#discussioncomment-3857114)
